operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -671,22 +672,7 @@ func (r *StackReconciler) Reconcile(ctx context.Context, request ctrl.Request) (
 	sess.SetSecretEnvs(ctx, stack.SecretEnvs, request.Namespace)
 
 	// Step 3: Evaluate whether an update is needed. If not, we transition to Ready.
-
-	// determine whether the stack is in sync with respect to the specification.
-	// i.e. the current spec generation has been applied, the update was successful,
-	// the latest commit has been applied, and (if resync is enabled) has been resynced recently.
-	resyncFreq := time.Duration(sess.stack.ResyncFrequencySeconds) * time.Second
-	if resyncFreq.Seconds() < 60 {
-		resyncFreq = time.Duration(60) * time.Second
-	}
-	synced := instance.Status.LastUpdate != nil &&
-		instance.Status.LastUpdate.Generation == instance.Generation &&
-		instance.Status.LastUpdate.State == shared.SucceededStackStateMessage &&
-		(isStackMarkedToBeDeleted ||
-			(instance.Status.LastUpdate.LastSuccessfulCommit == currentCommit &&
-				(!sess.stack.ContinueResyncOnCommitMatch || time.Since(instance.Status.LastUpdate.LastResyncTime.Time) < resyncFreq)))
-
-	if synced {
+	if isSynced(instance, currentCommit, isStackMarkedToBeDeleted) {
```

Comment:
nit: `isStackMarkedToBeDeleted` is redundant if you use `instance.DeletionTimestamp != nil`.